docs/adr/2025-11-07-process-lock.md

```
@@ -0,0 +1,175 @@
+#### Pros
+
+- Native database feature, no additional tables needed
+- Guaranteed cleanup on connection close
+- Simple implementation
+- No new infrastructure/dependencies (B7)
+- No change required in apps (B8)
+
+#### Cons
+
+- Critical issue: Requires holding a database connection open for the entire request duration (B2)
+    - Would exhaust connection pool quickly under moderate load if there are a high proportion of long-running requests
```

markdownlint-cli2 (0.18.1)

Unordered list indentation
Expected: 2; Actual: 4

(MD007, ul-indent)